src/script/event/EventService.ts

```diff
@@ -127,18 +127,52 @@ export class EventService {
 
     try {
       if (this.storageService.db) {
-        const entry = await this.storageService.db
+        return await this.storageService.db
+          .table(StorageSchemata.OBJECT_STORE.EVENTS)
+          .where('conversation')
+          .equals(conversationId)
+          .filter(item => item.data?.replacing_message_id === eventId || item.id === eventId)
+          .first();
+      }
+
+      const records = (await this.storageService.getAll(StorageSchemata.OBJECT_STORE.EVENTS)) as EventRecord[];
+      return records
+        .filter(record => record.id === eventId && record.conversation === conversationId)
+        .sort(compareEventsById)
+        .shift();
+    } catch (error) {
+      const logMessage = `Failed to get event '${eventId}' for conversation '${conversationId}': ${error.message}`;
+      this.logger.error(logMessage, error);
+      throw error;
+    }
+  }
+
+  /**
+   * Load a replacing event from database.
+   * To be used in certain cases (during some cases with quoting) when the original message cannot be found.
+   *
+   * @param conversationId ID of conversation
+   * @param eventId ID of event to retrieve
+   */
+  async loadReplacingEvent(conversationId: string, eventId: string): Promise<EventRecord> {
+    if (!conversationId || !eventId) {
+      this.logger.error(`Cannot get event '${eventId}' in conversation '${conversationId}' without IDs`);
+      throw new ConversationError(BASE_ERROR_TYPE.MISSING_PARAMETER, BaseError.MESSAGE.MISSING_PARAMETER);
+    }
+
+    try {
+      if (this.storageService.db) {
+        return await this.storageService.db
           .table(StorageSchemata.OBJECT_STORE.EVENTS)
           .where('id')
           .equals(eventId)
           .filter(record => record.conversation === conversationId)
           .first();
-        return entry;
       }
 
       const records = (await this.storageService.getAll(StorageSchemata.OBJECT_STORE.EVENTS)) as EventRecord[];
       return records
-        .filter(record => record.id === eventId && record.conversation === conversationId)
+        .filter(record => record.conversation === conversationId && record.data?.replacing_message_id === eventId)
```

Comment:
Let's **only** lookup by replacing id at this point. The consumer is responsible for falling back to something else if it's not found

Reply:
true

Reply:
in the other function on L140, we are also checking by conversation && then id. (This was also the previous implementation) Should we change that as well or leave both the same? 

I assume there was some logic to checking both, although the chances of an overlap here seem relatively astronomical to me. Still sort of unclear when this section of the function would be called anyway.